Gems/Atom/Tools/ShaderManagementConsole/Code/Source/ShaderManagementConsoleApplication.cpp

```diff
@@ -93,8 +95,11 @@ namespace ShaderManagementConsole
 
         // Overriding default document type info to provide a custom view
         auto documentTypeInfo = ShaderManagementConsoleDocument::BuildDocumentTypeInfo();
-        documentTypeInfo.m_documentViewFactoryCallback = [this](const AZ::Crc32& toolId, const AZ::Uuid& documentId) {
-            return m_window->AddDocumentTab(documentId, new ShaderManagementConsoleTableView(toolId, documentId, m_window.get()));
+        documentTypeInfo.m_documentViewFactoryCallback = [this](const AZ::Crc32& toolId, const AZ::Uuid& documentId)
+        {
+            auto* container = new QWidget;
+            new ShaderManagementConsoleContainer(container, toolId, documentId, m_window.get());
+            return m_window->AddDocumentTab(documentId, container);
         };
```

Comment:
What does the container widget do in this case?

Reply:
Oh yes, it serves to adapt the excepted pointer type. AddDocumenTab takes a QWidget*. I got this solution from a stackoverflow answer. The SMCContainer class derives from Layout* so it wasn't compatible without using this dummy intermediary.